### ISSUE TITLE
Clustering between datacenters

### ISSUE THREAD

**Issue by powerpbx** [irontec/ivozprovider#765]
Are there any suggestions for clustering over WAN between datacenters?  This appears to be the only relevant discussion.

https://github.com/irontec/ivozprovider/issues/271

**Comment by Kaian** [comment]
Hi @powerpbx 

> Have you guys testing going between datacenters using MySQL 5.7 master-master or group replication? Is that a viable solution between datacenters over WAN?

We have never clustered over WAN any of our production environments. All of them have a management private network for cluster, replications, monitoring and so on. The data nodes (that have mysql) usually doesn't even have a public IP.

> Percona or MariaDB 10 clustering seems to be the best solution. However, When we try replace MySQL 5.7 with Percona or MariaDB 10, we get dependency errors. Are there any suggestions how to get around this? Or perhaps there is something you can do with the repository to remove the dependency on MySQL 5.7.

We have already discussed this on #637. As we mentioned there, MySQL is ok for us during artemis release (all 2.x versions) as master-master over LAN replication should be enough. 
We won't change how our packages work for current release, but if you want us to add support for MariaDB, feel free to contact comercial@irontec.com for scheduling and pricings details for that change.

> We have tried to create our own repository for this purpose but it opened up a big can of worms. It is proving to be a challenge to maintain and keep synced with version changes happening on the upstream irontec git repository.

We also have custom debian packages for kamailio, asterisk, rtpengine, rtpproxy and more softwares. We choose some version and use them for a long time. We don't have the latest asterisk version, nor the latest packaged kamailio. Keeping sync with version changes of those projects is more than challenging.

Regards

**Comment by powerpbx** [comment]
So you are not datacenter redundant?   So you are prepared to accept the possibilility of an outage if the datacenter has problems?  We use a lot of different datacenters and they all have outages.  Even Amazon and Microsoft.  Not very often but it still happens sometimes.

**Comment by powerpbx** [comment]
So you are not datacenter redundant and are prepared to accept that you could have an outage if the datacenter has problems?  We use a lot of different datacenters and then all have outages sooner or later.  Including Amazon and Microsoft.  Not very often and usually not for very long but it still happens sometimes for one reason or another.

**Comment by Kaian** [comment]
Hi @powerpbx 

Yes, we are. But this Github project is for reporting software bugs, feature requests or project feedback.

Not sure what this issue is about right now:

- If you are requesting a package change, please follow previous note intructions to get information for commercial support. We will estimate the change and schedule it.

- If you are asking about how Irontec does their distributed deployments and clusters, then use the mentioned email to contact our inftrastructure team so you can ask them how deploys and replication in multiple datacenter works. That's out of scope for github project.

Our commercial team will we happy to help you if you need customizations or information.

Regards!!

**Comment by powerpbx** [comment]
I guess this would be a feature request if the ability doesn't already exist.  Documentation and features to achieve data center redundancy.

**Comment by Kaian** [comment]
I have tagged this as doc and feature and moved to Future milestone.

Take into account docummenting this is very low priority in our schedule, so this will happen after #271 and probably not anytime soon.

Regards and thanks for the feedback!

**Comment by powerpbx** [comment]
Have you guys tried putting Kamailio and RTP proxies behind NAT as well or should those be on public IP's?  The reason we ask is because in order to make those redundant and still be on LAN, we would need to use floating IP's and AWS uses NAT with floating IP's

**Comment by Kaian** [comment]
Sorry, we totally forgot this.

I highly doubt that floating IPs will work with our configuration files. We haven't tested enough the platform on cloud services, and NAT detection mechanisms most probably will only work on exposed machines with direct public IP addresses.

If you have any feedback of your tests or have successfully installed the platform on a cloud service (AWS, digitalocean, ...) feel free to add any useful information for other users.

Thanks @powerpbx !
